samples/AzCopyCore/AzCopyCore/StorageRequests.cs

```diff
@@ -102,23 +102,30 @@ public static ReadOnlyMemory<byte> AsString(Http.Method verb)
 
     public struct PutRangeRequest : IStorageRequest
     {
-        public Stream FileContent { get; set; } // TODO (pri 3): should there be a way to write from file handle or PipeReader?
-        public string FilePath { get; set; }
+        Stream _fileContent; // TODO (pri 3): should there be a way to write from file handle or PipeReader?
+        string _filePath;
+        long _offset;
+        int _length;
 
         // TODO (pri 3): I dont like how the client property is a public API
         public StorageClient Client { get; set; }
 
-        public PutRangeRequest(string filePath, Stream fileContent)
+        public PutRangeRequest(string filePath, Stream fileContent, long offset, int length)
         {
-            FilePath = filePath;
-            FileContent = fileContent;
+            if (offset < 0 || offset > fileContent.Length - length) throw new ArgumentOutOfRangeException(nameof(offset));
```

Comment:
Hmm. What about the case when `length > fileContent.Length`?

Reply:
AORE will be thrown. fileContent.Length - length will be negative and will be < offset (which has to be positive at the time of this check)